src/components/videocard/videocard.js

```diff
@@ -51,6 +53,16 @@ export default function Videocard({ video }) {
             navigate("/login");
         }
     }
+    const addToHistory = async () => {
+        if (user) {
+            const res = await axios.post("/api/user/history", { video }, {
+                headers: {
+                    authorization: encodedToken
+                }
+            });
+            dispatch({ type: "ADD_TO_HISTORY", payload: res.data.history });
+        }
```

Comment:
suggestion: you can define these types in a separate file say actions

Reply:
noted, will do this. thanks